b9lab-content/3-main-concepts/02-architecture.md

```diff
@@ -1 +1,294 @@
+
+Any application that uses Tendermint for consensus must implement the Application Blockchain Interface. Fortunately, you do not have to because the Cosmos SDK provides a boilerplate known as the BaseApp to get you started.
+
+In the following exercise, you will create a minimal distributed state machine with the Cosmos SDK. Your state machine will rely on Tendermint for consensus.
+
+## Long-running exercise
+
+Here, we introduce an exercise, a project, that you will bring forward as you learn more. In so doing, you will see how the Cosmos SDK improves your productivity by taking care of the boilerplate. It is still interesting to know about what the boilerplate is, in order to get a better overall picture.
+
+### The setup
+
+We are going to create a blockchain that lets interested people play checkers against each other. There are many versions of the rules. We choose [those](https://www.ducksters.com/games/checkers_rules.php). Also, the object of the exercise is to learn ABCI and Cosmos SDK, not to get lost in the proper implementation of the board state or of the rules of checkers. So we are going to use, and adapt, [this ready-made implementation](https://github.com/batkinson/checkers-go/blob/a09daeb/checkers/checkers.go), with the additional rule that the board is 8x8 and played on black cells. The code will likely require adaptations as we go along. Also, we are not going to be overly concerned with a marketable GUI since, again, that would be a separate project in itself. Of course, we still need to concern ourselves with creating the groundwork for a GUI to be possible in the first place.
+
+When we return to this long running exercise, in the next chapters, the goal is to improve it and use the Cosmos SDK as we learn elements.
+
+As you can imagine, beyond the rules of the game, there is a lot we need to take care of. Therefore, as a start we ought to simplify to the maximum. Have a look at these [specs](https://github.com/tendermint/spec/blob/c939e15/spec/abci/abci.md) to see what the application needs to do to comply with ABCI, and try to figure out how barebones you would make your first, imperfect, checkers game blockchain.
+
+### “Make” the state machine
+
+Here is how we would do it. We _would_ do it, because we are here to learn the Cosmos SDK primarily, not to become proficient at implementing ABCI.
+
+We want to show the minimum viable ABCI state machine. Tendermint does not concern itself with whether proposed transactions are valid, and about how the state changes after such a transaction. It delegates those to the state machine, which will _transform_ transactions into game moves and states.
```

Comment:
Perhaps we can kick this off with a flow-diagram to give an overview to a relatively heavy section?